### PR TITLE
More async refactoring Pt 3

### PR DESCRIPTION
**Pull request by MartinJibber**

### CODE REVIEW

File: Shared/Extensions/UI Extensions/UICollectionView+Extensions.swift [hunk 1]
```diff
@@ -31,5 +30,28 @@ extension UICollectionView {
             completion?(completed)
         })
     }
+
+    @discardableResult
+    func performBatchUpdates(modifyItems: (() -> Swift.Void)? = nil,
```
Comment (Benjamindodgson) [inline Shared/Extensions/UI Extensions/UICollectionView+Extensions.swift]: We should remove this. If its being used, it shouldnt be.

File: Shared/UI/Face Detection/ImageCaptureViewController.swift [hunk 2]
```diff
@@ -38,27 +38,30 @@ class ImageCaptureViewController: UIViewController, AVCaptureVideoDataOutputSamp
 
     private(set) var cameraType: CameraType = .front
     var flashMode: AVCaptureDevice.FlashMode = .auto
-#warning("Convert to async")
-    func requestAthorization() -> Future<Bool, Never> {
-        return Future { promise in
+
+    func requestAuthorization() async -> Bool {
+        let isGranted: Bool = await withCheckedContinuation { continuation in
             if self.isAuthorized {
```
Comment (Benjamindodgson) [inline Shared/UI/Face Detection/ImageCaptureViewController.swift]: Seems like I was redundant here. isGranted the same as isAuthorized? lets use just one.

Comment (Benjamindodgson) [reply Shared/UI/Face Detection/ImageCaptureViewController.swift]: Probably remove isAuthorized, I could see it getting out of sync.

File: iOS/Collection View Management /CollectionViewManager.swift [hunk 3]
```diff
@@ -68,28 +66,26 @@ class CollectionViewManager<SectionType: ManagerSectionType>: NSObject, UICollec
...
 
-            if let cycle = animationCycle {
-                self.animateOut(position: cycle.outToPosition, concatenate: cycle.shouldConcatenate) { [unowned self] in
-                    self.dataSource.apply(snapshot, animatingDifferences: animatingDifferences) {
-                        self.animateIn(position: cycle.inFromPosition,
-                                       concatenate: cycle.shouldConcatenate,
-                                       scrollToEnd: cycle.scrollToEnd) {
-                            self.hasLoadedInitialSnapshot = true
-                            promise(.success(()))
+            Task.onMainActor {
```
Comment (Benjamindodgson) [inline iOS/Collection View Management /CollectionViewManager.swift:73]: sexy.

File: iOS/Extensions/Twilio/TCHChannel+Extensions.swift [hunk 4]
```diff
@@ -25,20 +25,18 @@ extension TCHChannel {
...
-    func getNonMeMembers() -> Future<[TCHMember], Error> {
-        return Future { promise in
-            if let members = self.members?.membersList() {
-                var nonMeMembers: [TCHMember] = []
-                members.forEach { (member) in
-                    if member.identity != User.current()?.objectId {
-                        nonMeMembers.append(member)
-                    }
+
+    func getNonMeMembers() -> [TCHMember] {
```
Comment (Benjamindodgson) [inline iOS/Extensions/Twilio/TCHChannel+Extensions.swift:29]: TCHMember and a User share the same identifier. See if this is used anywhere and remove it. Or add warning todo later. We should only be modifying User objects.

**Review by Benjamindodgson — APPROVED** [review]

File: Shared/UI/Face Detection/ImageCaptureViewController.swift [hunk 5]
```diff
@@ -32,33 +32,18 @@ class ImageCaptureViewController: UIViewController, AVCaptureVideoDataOutputSamp
...
-    }
 
     func begin() {
-        self.requestAthorization()
-            .mainSink { authorized in
-                if authorized {
-                    self.configureCaptureSession()
-                    self.session.startRunning()
-                }
-            }.store(in: &self.cancellables)
```
Comment (MartinJibber) [inline Shared/UI/Face Detection/ImageCaptureViewController.swift:61]: Turns out we don't need any of these authorization convenience functions. There's now a built in async request access function.

**Pull request merged**